inference-engine/src/transformations/src/transformations/common_optimizations/random_uniform_fusion.cpp

```diff
@@ -0,0 +1,173 @@
+        ngraph::pattern::wrap_type<opset8::RandomUniform>({data_pattern, ru_min_input_pattern, ru_max_input_pattern},
+                                                          pattern::consumers_count(1));
+    const auto add_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+
+    const auto convert_pattern = ngraph::pattern::wrap_type<opset8::Convert>({random_uniform_pattern});
+    const auto convert_or_random_uniform_pattern =
+        std::make_shared<pattern::op::Or>(OutputVector{convert_pattern, random_uniform_pattern});
+    const auto add_pattern =
+        ngraph::pattern::wrap_type<opset8::Add>({convert_or_random_uniform_pattern, add_const_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto data = pattern_map.at(data_pattern);
+        const auto random_uniform = pattern_map.at(random_uniform_pattern);
+        const auto add_constant = pattern_map.at(add_const_pattern);
+        const auto ru_max_input = pattern_map.at(ru_max_input_pattern);
+        const auto ru_min_input = pattern_map.at(ru_min_input_pattern);
+        const auto add = pattern_map.at(add_pattern);
+        const auto ru = std::dynamic_pointer_cast<opset8::RandomUniform>(random_uniform.get_node_shared_ptr());
+
+        const auto add_const = std::dynamic_pointer_cast<opset8::Constant>(add_constant.get_node_shared_ptr());
+        if (!add_const)
+            return false;
+        if (!add_const->get_element_type().is_real())
+            return false;
+
+        auto const_shape = add_const->get_shape();
+        size_t const_shape_size = shape_size(const_shape);
+
+        if (const_shape_size != 1)
+            return false;
+        const auto value = add_const->cast_vector<double>();
+        auto new_add_const = op::Constant::create(ru->get_out_type(), Shape{}, value);
+
+        if (!ru)
+            return false;
+        if (!ru->get_out_type().is_real())
+            return false;
+
+        if (pattern_map.count(convert_pattern)) {
+            const auto& convert = pattern_map.at(convert_pattern);
+            const auto cvt = std::dynamic_pointer_cast<opset8::Convert>(convert.get_node_shared_ptr());
+            if (!cvt)
+                return false;
+            if (!cvt->get_element_type().is_real())
+                return false;
+            const auto add1 = std::make_shared<ngraph::opset8::Add>(ru_min_input, new_add_const);
+            const auto add2 = std::make_shared<ngraph::opset8::Add>(ru_max_input, new_add_const);
+            const auto new_ru = ru->clone_with_new_inputs({data, add1, add2});
+            new_ru->set_friendly_name(m.get_match_root()->get_friendly_name());
+            const auto new_ru_conv = cvt->clone_with_new_inputs({new_ru});
+            copy_runtime_info({add.get_node_shared_ptr(), ru, cvt}, {new_ru, new_ru_conv});
+            ngraph::replace_node(m.get_match_root(), new_ru_conv);
```

Comment:
Looks like the logic inside RandomUniformMulFusion and RandomUniformAddFusion are almost similar. Let's combine this two passes.

Reply:
Done.